untrusted/urts/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+#![allow(non_upper_case_globals)]
+
+use mc_sgx_urts_sys::{
+    sgx_create_enclave_from_buffer_ex, sgx_destroy_enclave, sgx_enclave_id_t, sgx_status_t,
+};
+use std::ops::Deref;
+use std::{os::raw::c_int, ptr};
+
+#[derive(PartialEq, Debug)]
+pub enum Error {
+    SgxStatus(sgx_status_t),
+}
+
+/// Struct for interfacing with the SGX SDK.  This should be used directly in
+/// sgx calls `ecall_some_function(*enclave, ...)`.
+///
+/// Avoid storing the de-referenced instance of the enclave.  The de-referenced
+/// value of the enclave will result in failures to the SGX SDK after the
+/// enclave is dropped.
+#[derive(Debug, PartialEq)]
+pub struct Enclave {
+    // The enclave ID, assigned by the sgx interface
+    id: sgx_enclave_id_t,
+}
+
+#[derive(Default)]
+pub struct EnclaveBuilder<'a> {
+    // The bytes for the enclave.
+    //
+    // This needs to be mutable per the signature of
+    // `sgx_create_enclave_from_buffer_ex()`.  See comment in
+    // [EnclaveBuilder::create()] for more info.
+    bytes: &'a mut [u8],
+
+    // `true` if the enclave should be created in debug mode
+    debug: bool,
+}
+
+impl<'a> EnclaveBuilder<'a> {
+    /// Returns an EnclaveBuilder for the provided signed enclave.
+    ///
+    /// # Arguments
+    ///
+    /// * `bytes` - The bytes representing the enclave file.  This should be a
+    ///     signed enclave.  This needs to be mutable as the sgx interface will
+    ///     modify the bytes in [EnclaveBuilder::create()].  The `bytes` will
+    ///     not be usable to create another enclave after calling
+    ///     [EnclaveBuilder::create()].  If more than one enclave is needed from
+    ///     the same original `bytes` be sure and copy them to each builder,
+    ///     prior to calling [EncalveBuilder::create()].
+    pub fn new(bytes: &mut [u8]) -> EnclaveBuilder {
+        EnclaveBuilder {
+            bytes,
+            ..Default::default()
```

Comment:
Let's spell out all the fields here
```suggestion
            debug: false,
```